xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -5654,3 +5660,78 @@ void CVideoPlayer::SetUpdateStreamDetails()
 {
   m_messenger.Put(std::make_shared<CDVDMsg>(CDVDMsg::PLAYER_SET_UPDATE_STREAM_DETAILS));
 }
+
+
+void CVideoPlayer::NotifySubtitleUpdate(bool bStatus, bool bStreamInfo)
+{
+  CVariant data;
+  data["player"]["playerid"] = static_cast<int>(CServiceBroker::GetPlaylistPlayer().GetCurrentPlaylist());
+  if (bStatus)
+  {
+    data["property"]["subtitleenabled"] = m_processInfo->GetVideoSettings().m_SubtitleOn;
+  }
+  if (bStreamInfo)
+  {
+    int iStream = m_processInfo->GetVideoSettings().m_SubtitleStream;
+    SubtitleStreamInfo info;
+    GetSubtitleStreamInfo(iStream, info);
+    if (info.valid)
+    {
+      CVariant contentEntry(CVariant::VariantTypeObject);
+      contentEntry["index"] = iStream;
```

Comment:
Things seem a bit mixed up, or is it on purpose? No notification with subtitleenabled=false when subtitles are disabled?
```suggestion
  data["property"]["subtitleenabled"] = m_processInfo->GetVideoSettings().m_SubtitleOn;
    
  if (bStatus && bStreamInfo)
  {
    const int stream = m_processInfo->GetVideoSettings().m_SubtitleStream;
    SubtitleStreamInfo info{};
    GetSubtitleStreamInfo(stream, info);
    if (info.valid)
    {
      CVariant contentEntry(CVariant::VariantTypeObject);
      contentEntry["index"] = stream;
      ... other properties and addition to data
    }
  }
  CServiceBroker::GetnnouncementManager()->Announce(ANNONOUNCEMENT::Player, "OnPropertyChanged", data);
 }
```